Wikipedia/Code/WikidataDescriptionEditingController.swift

```diff
@@ -0,0 +1,98 @@
+public struct WikidataAPI {
+    public static let host = "www.wikidata.org"
+    public static let path = "/w/api.php"
+    public static let scheme = "https"
+
+    public static var urlWithoutAPIPath: URL? {
+        var components = URLComponents()
+        components.scheme = scheme
+        components.host = host
+        return components.url
+    }
+}
+
+struct WikidataAPIResult: Decodable {
+    struct Error: Decodable {
+        let code, info: String?
+    }
+    let error: Error?
+    let success: Int?
+}
+
+extension WikidataAPIResult.Error: LocalizedError {
+    var errorDescription: String? {
+        return info
+    }
+}
+
+extension WikidataAPIResult {
+    var succeeded: Bool {
+        return success == 1
+    }
+}
+
+@objc public final class WikidataDescriptionEditingController: NSObject {
+    private var blacklistedLanguages = Set<String>()
+
+    @objc public func setBlacklistedLanguages(_ blacklistedLanguagesFromRemoteConfig: Array<String>) {
+        blacklistedLanguages = Set(blacklistedLanguagesFromRemoteConfig)
+    }
+
+    public func isBlacklisted(_ languageCode: String) -> Bool {
+        guard !blacklistedLanguages.isEmpty else {
+            return false
+        }
+        return blacklistedLanguages.contains(languageCode)
+    }
+
+    @objc(publishNewWikidataDescription:forArticle:completion:)
+    public func publish(newWikidataDescription: String, for article: MWKArticle, completion: @escaping (_ error: Error?) -> Void) {
+        guard let title = article.displaytitle,
```

Comment:
`displaytitle` can contain html so i think we need `article.url.wmf_title` here...

noticed when trying to edit description of https://en.wikipedia.org/wiki/Ashurbanipal_(sculpture)
`displaytitle` : `<i>Ashurbanipal</i> (sculpture)`
`article.url.wmf_title` : `Ashurbanipal (sculpture)`